### PR TITLE
feat(api): add handling empty metadata suggestions

### PR DESCRIPTION
**Pull request by JakubKida**
Throw an `ERROR_CODE_EMPTY_METADATA_SUGGESTIONS` error on receiving no AI suggestions for metadata.

### CODE REVIEW

**Review by greg-in-a-box — APPROVED** [review]

File: src/api/Metadata.js [hunk 1]
```diff
@@ -28,6 +29,7 @@ import {
     ERROR_CODE_FETCH_METADATA_TEMPLATES,
     ERROR_CODE_FETCH_SKILLS,
     ERROR_CODE_FETCH_METADATA_SUGGESTIONS,
+    ERROR_CODE_EMPTY_METADATA_SUGGESTION,
```
Comment (greg-in-a-box) [inline src/api/Metadata.js:32]:
```suggestion
    ERROR_CODE_EMPTY_METADATA_SUGGESTIONS,
```
Strange this didnt break before you pushed up

File: src/api/__tests__/Metadata.test.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ import {
     METADATA_TEMPLATE_CLASSIFICATION,
     METADATA_TEMPLATE_PROPERTIES,
     TYPE_FILE,
+    ERROR_CODE_EMPTY_METADATA_SUGGESTION,
```
Comment (greg-in-a-box) [inline src/api/__tests__/Metadata.test.js:20]:
```suggestion
    ERROR_CODE_EMPTY_METADATA_SUGGESTIONS,
```

**Review by greg-in-a-box — CHANGES_REQUESTED** [review]
I almost missed the fixed time i reviewed it, looks like `ERROR_CODE_EMPTY_METADATA_SUGGESTION` should be plural in the .js and the test file because the constants is `ERROR_CODE_EMPTY_METADATA_SUGGESTIONS`

File: src/api/__tests__/Metadata.test.js [hunk 3]
```diff
@@ -578,10 +579,7 @@ describe('api/Metadata', () => {
             metadata.getEditors = jest.fn().mockResolvedValueOnce('editors');
             metadata.getCustomPropertiesTemplate = jest.fn().mockReturnValueOnce('custom');
             metadata.getUserAddableTemplates = jest.fn().mockReturnValueOnce('templates');
-            metadata.getTemplates = jest
-                .fn()
-                .mockResolvedValueOnce('global')
-                .mockResolvedValueOnce('enterprise');
+            metadata.getTemplates = jest.fn().mockResolvedValueOnce('global').mockResolvedValueOnce('enterprise');
```
Comment (wpiesiak) [inline src/api/__tests__/Metadata.test.js:582]: Are those formatting changes necessary?

Comment (JakubKida) [reply src/api/__tests__/Metadata.test.js:582]: Prettier changed them by default, not sure if I have it set up wrong or were the latest guidelines not applied to this file

**Review by wpiesiak — APPROVED** [review]

**Review by greg-in-a-box — APPROVED** [review]

**Review by JChan106 — APPROVED** [review]

**Pull request merged**